backend/src/GraphQl/MethodAsService/FileInputData.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+
+namespace Database.GraphQl.MethodAsService;
+
+public sealed record FileInputData(
+    Data Data
+);
+
+public sealed record Data(
+    ComponentCharacteristics ComponentCharacteristics,
+    IReadOnlyList<DataPoint> DataPoints
+);
+
+public sealed record ComponentCharacteristics(
+    string Surface,
+    string Symmetries
+);
+
+public sealed record DataPoint(
+    Incidence Incidence,
+    Emergence Emergence,
+    Results Results
+);
+
+public sealed record Incidence(
+    Wavelengths Wavelengths,
+    Direction Direction
+);
+
+public sealed record Emergence(
+    Direction Direction
+);
+
+public sealed record Results(
+    double Transmittance,
+    double? Reflectance = null,
+    ColorTransmittance? ColorTransmittance = null,
+    ColorReflectance? ColorReflectance = null
+);
+
+public sealed record Direction(
+    int Polar
+);
+
+public sealed record Wavelengths(
+    int Wavelength
+);
+
+public record ColorTransmittance(
+    IReadOnlyList<CielabColor> CielabColors,
+    IReadOnlyList<ColorRenderingIndex> ColorRenderingIndices
+);
+
+public record ColorReflectance(
+    IReadOnlyList<CielabColor> CielabColors
+);
+
+public record CielabColor(
+    int LStar,
+    double AStar,
+    double BStar,
+    string Observer,
```

Comment:
Observer and illuminant are not strings but enumerations, see https://github.com/building-envelope-data/api/blob/48fca7aef7cf813d4a63b7a8497409448fff6406/schemas/opticalData.json#L534-L543